hack/update-generated-docs.sh

```diff
@@ -25,6 +25,9 @@ set -o pipefail
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
+# This only runs on master
+[[ $(git rev-parse --abbrev-ref HEAD) == "master" ]] || exit
```

Comment:
Why? Won't this disable the script for people doing work in dev branches?

Is this script doing something wrong in release branches?